manual: use subheadings for primitive types (#6846)
NixOS/nix: doc/manual/src/expressions/language-values.md

Note that lists are only lazy in values, and they are strict in length.

> ## Attribute Set
> 
> Attribute sets are collections of name-value-pairs (called *attributes*) enclosed in curly brackets (`{ }`).

Reviewer: If you're de-pluralizing the heading, maybe the body should reflect that as well? (There may be more examples out of view of the diff.)

e.g.:
```suggestion
## Attribute Set

An attribute set is a collection of name-value-pairs (called *attributes*) enclosed in curly brackets (`{ }`).
```
